src/components/TextEditor/components/TextEditor.js

```diff
@@ -1,164 +1,151 @@
-import { faDownload, faSave, faShare, faCodeBranch } from '@fortawesome/free-solid-svg-icons';
+import {
+  faDownload, faSave, faShare, faCodeBranch,
+} from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import ReactModal from 'react-modal';
 import { Redirect } from 'react-router-dom';
 import { Button } from 'reactstrap';
 import { EDITOR_WIDTH_BREAKPOINT, ThumbnailArray } from '../../../constants';
-import sketch from '../../../lib';
-import * as fetch from '../../../lib/fetch.js';
-
-import DropdownButtonContainer from '../containers/DropdownButtonContainer';
+import skt from '../../../lib';
+import * as fetch from '../../../lib/fetch';
 import OpenPanelButtonContainer from '../../common/containers/OpenPanelButtonContainer';
-import ViewportAwareButton from '../../common/ViewportAwareButton.js';
-import EditorRadio from './EditorRadio.js';
+import ViewportAwareButton from '../../common/ViewportAwareButton';
+import DropdownButtonContainer from '../containers/DropdownButtonContainer';
+import EditorRadio from './EditorRadio';
 import ShareSketchModal from './ShareSketchModal';
-
+/* eslint-disable no-shadow */
+/* eslint-disable func-names */
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-use-before-define */
+/* eslint-disable no-unused-expressions */
```

Comment:
we can chat about this during the meeting today! i hope i have time to come over, or feel free to come a bit earlier! im here already